pkg/nightly/nightly_test.go

```diff
@@ -122,16 +136,26 @@ func TestSingleDC(t *testing.T) {
 			clusterName := makeClusterName(t)
 			defer destroyCluster(t, l, clusterName)
 
-			err = runCmds(l, [][]string{
+			if err := runCmds(ctx, l, [][]string{
 				{"roachprod", "create", clusterName},
 				{"roachprod", "put", clusterName, *cockroach, "./cockroach"},
 				{"roachprod", "put", clusterName, *workload, "./workload"},
 				{"roachprod", "start", clusterName},
-				{"roachprod", "ssh", clusterName + ":1", "--", testCmd},
-			})
-			if err != nil {
+			}); err != nil {
 				t.Fatal(err)
 			}
+
+			m := monitorWithContext(ctx)
+			m.Worker(func(ctx context.Context) error {
+				// TODO(dan): runCmd internally uses exec.CommandContext, which
+				// is supposed to kill the exec'd process if the context is
+				// canceled. This appears to work generally, but not for
+				// `roachprod ssh`.
+				return runCmd(ctx, l, "roachprod", "ssh", clusterName+":1", "--", testCmd)
```

Comment:
I think I fixed this in cockroachdb/roachprod#52.

Reply:
Same problem.

Reply:
Womp womp.